tsconfig.json

```diff
@@ -0,0 +1,47 @@
+{
+  "compilerOptions": {
+    "baseUrl": ".",
+    "paths": {
+      "server/*": [
+        "server/*"
+      ],
+      "client/*": [
+        "client/*"
+      ]
+    },
+    "allowJs": true,
+    "allowSyntheticDefaultImports": true,
+    "jsx": "preserve",
+    "lib": [
+      "dom",
+      "es2017"
+    ],
+    "module": "esnext",
```

Comment:
If we decide to go for Target: `es5`, then Module should be set to `commonjs`

Reply:
Unfortunately esnext is required for next and enforced by next as well (it literally overwrites tsconfig) - I made a separate tsconfig overriding this for the server since node requires commonjs.

Relevant thread: https://github.com/zeit/next.js/issues/7361